utils/utils.go

```diff
@@ -57,12 +58,34 @@ func DetermineNodename(conf types.NetConf) string {
 		nodename = conf.Hostname
 		logrus.Warn("Configuration option 'hostname' is deprecated, use 'nodename' instead.")
 	}
+	if nff := nodenameFromFile(); nff != "" {
+		logrus.Debugf("Read node name from file: %s", nff)
+		nodename = nff
+	}
 	if conf.Nodename != "" {
+		logrus.Debugf("Read node name from CNI conf: %s", conf.Nodename)
 		nodename = conf.Nodename
 	}
+	logrus.Debugf("Using node name %s", nodename)
 	return nodename
 }
 
+// nodenameFromFile reads the /var/lib/calico/nodename file if it exists and
+// returns the nodename within.
+func nodenameFromFile() string {
```

Comment:
A nodeNameFile that exists but is empty takes an interesting path through this code, but I suppose it is ok...

Reply:
Hm, yeah. I suspect we probably want to error in that case.

Reply:
Actually, let's leave this as it is for now. I think the code will do the right thing and I'm not sure how we'd end up with an empty file anyway.